src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import duke.command.Command;
+
+/**
+ * Represents the main entry point of the Duke application.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initialises a Duke application with UI and storage.
+     * @param filePath Path of file to store created tasks.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts running this application, accepting and responding to inputs from user.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```

Comment:
I like how you named the boolean variable in this way to make it sounds like a boolean 👍